### DISCUSSION TITLE
Issues with punctuation mark

### DISCUSSION THREAD

**Discussion by Fish31520** [Nachtzuster/BirdNET-Pi#64] [Q&A] [answered]
Hi,

I switched from Mcguirepr89 to Nachtzuster's version few weeks ago. So far everything seems to work smoothly but this morning I realized that birds with an apostrophe (Verdier d'Europe for example) were not recorded.

@Nachtzuster Do you think you could take a look at that ?

Thanks in advance for your help.

Jean

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/112718877/1e18f75a-e885-4a5b-beed-3c381811b9fa)
![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/112718877/8f76258d-cbc8-4c0a-889e-1014c78b7673)
![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/112718877/3f2aa3ed-15f3-4a05-b89a-43c14acd669b)
![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/112718877/4992a306-b6be-4275-8e7c-12fb33dfff48)

**Comment by Nachtzuster** [comment, marked as answer]
hmmm, are you running the latest version? Tools->System Controls->Update

**Reply by Fish31520** [reply]
Oops, nope, I had 15 updates to do...

My bad... 😅

**Comment by lloydbayley** [comment]
That's weird, this was hashed out in #41, so it would be interesting if it was still a problem.

**Reply by Fish31520** [reply]
I updated my system, I'll check if it solves my issues.

Thanks !

**Reply by lloydbayley** [reply]
Good luck! :)

**Reply by lloydbayley** [reply]
Any joy? Did it resolve things?

**Reply by Fish31520** [reply]
Hi,

So far everything seems to work perfectly !

Thanks once again !

Cheers !

**Reply by lloydbayley** [reply]
Fantastic news! Thanks for reporting back! :) We can close this now with a warm fuzzy feeling! :)

**Comment by Fish31520** [comment]
@Nachtzuster @lloydbayley 

Hi,

I'm reopening this topic cause this morning I realized I still had issues with a ponctuation mark, this time with the apostrophe.

Birds like the "Loriot d'Europe" or the "Râle d'eau" are correctly identified but there seems to be no recordings for them.

![image](https://github.com/user-attachments/assets/b50884d2-a672-4269-9f0b-3f505febfe40)

![image](https://github.com/user-attachments/assets/98f5d878-e634-4924-a306-a5761f628218)

I updated my system just after discovering that issue but it doesn't seems to solve it.

Any clue ?

**Comment by Nachtzuster** [comment]
Are the recordings available via `Recordings->By Species`?

**Reply by Fish31520** [reply]
Hi ! 

Strangely they are available from the `Recordings/By species` path but not in the `Best recordings` part.

![image](https://github.com/user-attachments/assets/c9fa2d69-b558-4ac0-85f7-ded4ffb908e2)

![image](https://github.com/user-attachments/assets/3cf03640-d5ca-4ee1-8c06-e5c39d240054)

**Comment by Nachtzuster** [comment]
@Fish31520 could you test the fix_apostrophe branch to see if it fixes the issue:
```
git remote remove origin
git remote add origin https://github.com/Nachtzuster/BirdNET-Pi.git
git fetch
./scripts/update_birdnet.sh -b fix_apostrophe
```

**Reply by Fish31520** [reply]
@Nachtzuster 

Sorry for that late answer.

I did install the new branch and rebooted, but it doesn't seems to fix the issue, recordings are still available in the `Recordings` part but not in the `Best Recordings` one...

**Reply by Nachtzuster** [reply]
I only did a change in the 'Species Stats' page, so that does make sense.

To be clear: the problem is still present in the 'Species Stats' page?

**Reply by Fish31520** [reply]
Stats are goods but the last recording is not available on that page.

![image](https://github.com/user-attachments/assets/72db6751-7fc7-4b8c-af32-c851f748847e)

![image](https://github.com/user-attachments/assets/e2ef437a-e8cd-4605-a608-caf189f7ff1f)

**Reply by Nachtzuster** [reply]
ok, thanks - so my hunch was incorrect it seems. this will need some actual debugging

**Reply by Fish31520** [reply]
@Nachtzuster 

You're welcome, thanks for your help and your patience !

I'll be AFK for 2 weeks but let me know if you need me to check anything else and I'll take a look when I'll be back.

Jean

**Reply by Nachtzuster** [reply]
@Fish31520 I pushed two fixes to the main branch, should be properly fixed now :-)
update from the UI and you'll switch back to the main branch.

**Reply by Nachtzuster** [reply]
@alexbelgium FYI fix affecting especially French.

**Reply by Fish31520** [reply]
@Nachtzuster
Thanks a lot !

**Reply by Fish31520** [reply]
@Nachtzuster 

I'm back to my keyboard ! 😄

Everything seems to be fine thanks to your fix, everything works perfectly !

Thanks a lot !
